django/db/migrations/operations/models.py

```diff
@@ -374,10 +374,9 @@ def database_forwards(self, app_label, schema_editor, from_state, to_state):
             old_db_table = old_model._meta.db_table
             new_db_table = new_model._meta.db_table
             # Don't alter when a table name is not changed.
-            if old_db_table == new_db_table:
-                return
-            # Move the main table
-            schema_editor.alter_db_table(new_model, old_db_table, new_db_table)
+            if old_db_table != new_db_table:
+                # Move the main table
+                schema_editor.alter_db_table(new_model, old_db_table, new_db_table)
```

Comment:
I don't see much value in keeping this. `BaseDatabaseSchemaEditor.alter_db_table()` is already a noop when a table name is not changed. The whole point of afeafd6036616bac8263d762c1610f22241c0187 was in avoiding altering related fields, but we cannot do this.  

I'd revert afeafd6036616bac8263d762c1610f22241c0187 and added a new test.

Reply:
I agree. Yesterday I was stuck trying to pass the noop test, but it'd be very ugly and probably breaking other corner cases.

Reply:
@iurisilvio Do you have time to keep working on this? If not I can push edits according to the https://github.com/django/django/pull/15993#discussion_r954605502.

Reply:
Yes, I'll update it today!